ygen/gogen.go

```diff
@@ -543,8 +555,48 @@ func makeTemplate(name, src string) *template.Template {
 // generation was targeted, along with the includePaths indicating where
 // imported modules were sourced from.
 func writeGoHeader(yangFiles, includePaths []string, cfg GeneratorConfig) (string, error) {
+
+	// Determine the running binary's name.
+	if cfg.Caller == "" {
+		cfg.Caller = callerName()
+	}
+
+	if cfg.PackageName == "" {
+		cfg.PackageName = defaultPackageName
+	}
+
+	if cfg.GoOptions.YgotImportPath == "" {
+		cfg.GoOptions.YgotImportPath = DefaultYgotImportPath
```

Comment:
Add something like "If cfg.GoOptions.YgotImportPath is empty, it is set to DefaultYgotImportPath." in the function description, for this and other defaults.